nvtabular/torch_dataloader.py

```diff
@@ -222,7 +222,45 @@ def __del__(self):
         self.buff.stop()
 
 
-class AsyncTensorBatchDatasetItr(torch.utils.data.IterableDataset):
+class TorchAsyncItr(torch.utils.data.IterableDataset):
```

Comment:
Why do we need this class in addition to the AsyncTensorBatchDatasetItr class? This class is a thin wrapper over AsyncTensorBatchDatasetItr which in turn is a thin wrapper over AsyncIterator - can we remove some of this indirection and have this use AsyncIterator directly?

Reply:
The idea is that the AsyncTensorBatchDatasetItr  provides the generic iterator and does the factory stuff... the initial reason for this was to remove the requirement of the torch.utils.data.iterabledataset from the base components of the nvt dataloading logic. this way you just create a wrapper and the tensor iterator in the factory and you can easily use the rest of the functionality while having support for target framework.

Reply:
I guess what I was trying to ask is whats the difference between AsyncIterator and AsyncTensorBatchDatasetItr now that you've removed the torch compatibility from AsyncTensorBatchDatasetItr. (like what do we need AsyncTensorBatchDatasetItr for anymore - it doesn't seem to implement any functionality ?)

Reply:
you are absolutely right I can just factor that out. Good catch!

Reply:
Really good... yeah this PR is useless. I will close.